git-repository/src/repository.rs

```diff
@@ -81,7 +72,36 @@ pub mod open {
                     crate::path::is_git(&git_dir).map(|kind| (git_dir, kind))?
                 }
             };
-            crate::Path::from_dot_git_dir(path, kind).try_into().map_err(Into::into)
+            let (git_dir, worktree_dir) =
+                crate::Path::from_dot_git_dir(path, kind).into_repository_and_work_tree_directories();
+            Repository::open_from_paths(git_dir, worktree_dir)
+        }
+
+        pub(in crate::repository) fn open_from_paths(
+            git_dir: PathBuf,
+            mut worktree_dir: Option<PathBuf>,
+        ) -> Result<Self, Error> {
+            if worktree_dir.is_none() {
+                let config = git_config::file::GitConfig::open(git_dir.join("config"))?;
+                let is_bare = config
+                    .value::<Boolean<'_>>("core", None, "bare")
+                    .map_or(false, |b| matches!(b, Boolean::True(_)));
```

Comment:
@edward-shen I hope you are doing well!

Now that I am using `git-config` for the first time, I could imagine accessing values more comfortably. Unless I am missing something, are there plans to make this more ergonomic?
I could probably add what would help me with more convenient access, but love to align with you first. Thank you

Reply:
That's the intent of ResolvedGitConfig. The current GitConfig was just a lightweight wrapper around the parser. I think I haven't written any accessor method on it, but that should be an easy add.

As for when I can get work done, I'm currently in the process of a cross continent move (NY -> CA), so unfortunately I don't know when I can work on it.

Reply:
> That's the intent of ResolvedGitConfig. The current GitConfig was just a lightweight wrapper around the parser. I think I haven't written any accessor method on it, but that should be an easy add.

With this I am happy to wait, and if you have ideas on how this API should look like, I'd be happy to add these methods as I go.